src/components/app/details/appDetails/AppStatusModal.tsx

```diff
@@ -0,0 +1,135 @@
+import React, { useEffect, useState } from 'react';
+import { ReactComponent as Close } from '../../../../assets/icons/ic-close.svg';
+import { ReactComponent as Error } from '../../../../assets/icons/ic-error-exclamation.svg';
+import { AppStreamData, AggregatedNodes } from '../../types';
+import { Drawer } from '../../../common';
+
+export const NodeStatusSortOrder = {
+    "progressing": 5,
+    "running": 6,
+    "healthy": 7,
+}
+
+export const AppStatusModal: React.FC<{
+    streamData: AppStreamData;
+    nodes: AggregatedNodes;
+    appName: string;
+    environmentName: string;
+    status: any;
+    message: string;
+    close: (...args) => void;
+}> = ({ streamData, nodes, status, close, message, appName, environmentName }) => {
+
+    const [nodeStatusMap, setNodeStatusMap] = useState(new Map());
+    const [rows, setRows] = useState([]);
+
+    useEffect(() => {
+        const stats = streamData?.result?.application?.status?.operationState?.syncResult?.resources?.reduce(
+            (agg, curr) => {
+                agg.set(`${curr.kind}/${curr.name}`, curr);
+                return agg;
+            },
+            new Map(),
+        );
+        setNodeStatusMap(stats);
+    }, [streamData]);
+
+    useEffect(() => {
+        let allRows = [];
+        let arr = Object.values(nodes.nodes).map(nodeMap => { return [...(nodeMap as Map<any, any>).values()] });
+        allRows = arr.reduce((acc, array) => {
+            acc = acc.concat(array)
+            return acc;
+        }, [])
+        allRows = allRows.filter(node => node.kind.toLowerCase() !== "rollout");
+        allRows = allRows.map(node => {
+            let nodeStatus = node?.status?.toLowerCase() || node?.health?.status?.toLowerCase();
+            let nodeStatusSortOrder = 1;
+            if (!nodeStatus) nodeStatusSortOrder = 100;
```

Comment:
use curly braces, dont put block in same line as if and else

Reply:
can follow the logic of sort order and how is it used?

Reply:
Sorted array of nodes based on key `nodeStatusSortOrder`.

- anything that is not running, healthy, progressing, null and empty -> we assume it to be error and place it at top -> nodeStatusSortOrder = 1
- progressing -> place it before running/healthy -> nodeStatusSortOrder = 5
- running/healthy -> place it at before null or empty -> nodeStatusSortOrder = 6/7
- null or empty -> place it at bottom -> nodeStatusSortOrder = 100

have following const:
 `const NodeStatusSortOrder = {
    "progressing": 5,
    "running": 6,
    "healthy": 7,
}`

Reply:
understood